### DISCUSSION TITLE
Add elements in Dynamic View

### DISCUSSION THREAD

**Discussion by lloydfischer** [likec4/likec4#967] [Q&A]
First, my thanks for likeC4. 

I'm trying to emulate the deployment view in structurizr and it seems like a Dynamic view is what I want. The key to the deployment view is that is contains elements and (maybe) relationships that are not in the model but exist only in the view. The documentation for Dynamic Views hints at this with "with specific elements and interactions, defined only in the view" and I can create indeed create relationships. I get a parse error creating an element. 

Am I missing something, or is this not yet implemented? 

This particular feature is vital in my use case, otherwise all the infrastructure details get mixed in with the main architecture. I could create these infra elements in the model, but then something like an  instanceof operator would be required.  

Again, I think likeC4 is a giant leap forward in this space, thanks so much

**Comment by lloydfischer** [comment]
I just realized that this discussion is a near duplicate of the #533 . Both are inquiring about supporting something like the Structurizr deployment construct.

**Comment by dstaas** [comment]
I don't believe it is possible to scope elements within a dynamic view, only relationships.  I agree this would be a very useful addition, since today what I'm doing is adding the elements to the 'big' model even for a dynamic view.  I would love the ability to scope both elements and relationships to a dynamic view.

My thanks to the authors - this is a very useful framework and the work is outstanding.
